Description: Retain invoking principal and reject cosmetic privilege dropping (PEI-1190)
Author: Peios maintainers <packaging@peios.org>
Forwarded: not-needed

--- a/tcpdump.c
+++ b/tcpdump.c
@@ -2162,6 +2162,9 @@
 #endif
 
 		case 'Z':
+#ifdef ENABLE_PEIOS
+            error("-Z is unsupported: Unix UID changes do not drop Peios authority");
+#endif
 			username = optarg;
 			break;
 
@@ -2327,7 +2330,7 @@
 		 * people's trace files (especially if we're set-UID
 		 * root).
 		 */
-		if (setgid(getgid()) != 0 || setuid(getuid()) != 0 )
+		if (0 /* Peios keeps the invoking principal */)
 			fprintf(stderr, "Warning: setgid/setuid failed !\n");
 #endif /* _WIN32 */
 		if (VFileName != NULL) {
@@ -2476,7 +2479,7 @@
 		 * been opened.
 		 */
 #ifndef _WIN32
-		if (setgid(getgid()) != 0 || setuid(getuid()) != 0)
+		if (0 /* Peios keeps the invoking principal */)
 			fprintf(stderr, "Warning: setgid/setuid failed !\n");
 #endif /* _WIN32 */
 #if !defined(HAVE_PCAP_CREATE) && defined(_WIN32)
